numba_dpex/ocl/oclimpl.py

```diff
@@ -286,20 +229,15 @@ def native_atomic_add(context, builder, sig, args):
 @lower(stubs.atomic.add, types.Array, types.UniTuple, types.Any)
 @lower(stubs.atomic.add, types.Array, types.Tuple, types.Any)
 def atomic_add_tuple(context, builder, sig, args):
-    device_type = dpctl.get_current_queue().sycl_device.device_type
     dtype = sig.args[0].dtype
 
-    if dtype == types.float32 or dtype == types.float64:
-        if (
-            device_type == dpctl.device_type.gpu
-            and config.NATIVE_FP_ATOMICS == 1
-        ):
-            return native_atomic_add(context, builder, sig, args)
-        else:
-            # Currently, DPCPP only supports native floating point
-            # atomics for GPUs.
-            return atomic_add(context, builder, sig, args, "add")
-    elif dtype == types.int32 or dtype == types.int64:
+    # TODO: do we need this check, or should we just use native_atomic_add for everything?
```

Comment:
The check should check if atomic operations are supported for the type, and it should be based on what SYCL supports: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atomic-references

Reply:
Does something like this work:
```python
def support_atomic(dtype: types.Type) -> bool:
    has_atomic64_support = True # TODO: get from dpctl

    # This check should be the same as described in sycl documentation:
    # https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atomic-references
    return (dtype == types.int32
    or dtype == types.uint32
    or dtype == types.float32
    or (dtype == types.int64
    or dtype == types.uint64
    or dtype == types.float64)
    and has_atomic64_support)
```